src/util.c

```diff
@@ -627,6 +627,93 @@ int d2string(char *buf, size_t len, double value) {
     return len;
 }
 
+/* Convert a double into a string with 'fractional_digits' digits after the dot precision.
+ * This is an optimized version of snprintf "%.<fractional_digits>f".
+ */
+int fixedpoint_d2string(char *dst, size_t dstlen, double dvalue, int fractional_digits) {
+    // min size of 2 ( due to 0. ) + n fractional_digitits + \0
+    if ((int)dstlen < (fractional_digits+3))
+        goto err;
+    if (dvalue == 0) {
+        dst[0] = '0';
+        dst[1] = '.';
+        for (int i = 0; i < fractional_digits; i++) {
+            dst[2 + i] = '0';
+        }
+        dst[fractional_digits+3] = '\0';
+        return fractional_digits + 2;
+    }
+    // scale and round
+    long long svalue = llrint(dvalue * pow(10.0, (double)fractional_digits));
+    unsigned long long value;
+    // write sign
+    int negative = 0;
+    if (svalue < 0) {
+        if (svalue != LLONG_MIN) {
+            value = -svalue;
+        } else {
+            value = ((unsigned long long) LLONG_MAX)+1;
+        }
+        if (dstlen < 2)
+            goto err;
+        negative = 1;
+        dst[0] = '-';
+        dst++;
+        dstlen--;
+    } else {
+        value = svalue;
+    }
+
+    static const char digitsd[201] =
```

Comment:
This algorithm is taken from another place, let's add a short comment with a reference